commit 6ddcd46463aa66855f1f8e74de454740a5f4aef4
Author: Avinash Patil <patila@marvell.com>
Date:   Mon Jun 11 18:13:57 2012 -0700

    mwifiex: fix incorrect privacy setting in beacon and probe response
    
    Test procedure:
    1. Start AP with security setting (e.g. WPA2)
    2. Stop AP
    3. Start AP with open security
    
    Here it's observed that privacy is enabled in beacons and
    probe responses.
    
    This patch fixes it by checking the privacy parameter from
    cfg80211_ap_settings. If privacy is not set in cfg80211_ap_settings,
    set open authentication and no encryption in FW.
    
    Signed-off-by: Avinash Patil <patila@marvell.com>
    Signed-off-by: Bing Zhao <bzhao@marvell.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/mwifiex/uap_cmd.c b/drivers/net/wireless/mwifiex/uap_cmd.c
index 8173ab66066d..89f9a2a45de3 100644
--- a/drivers/net/wireless/mwifiex/uap_cmd.c
+++ b/drivers/net/wireless/mwifiex/uap_cmd.c
@@ -27,6 +27,17 @@ int mwifiex_set_secure_params(struct mwifiex_private *priv,
 			      struct cfg80211_ap_settings *params) {
 	int i;
 
+	if (!params->privacy) {
+		bss_config->protocol = PROTOCOL_NO_SECURITY;
+		bss_config->key_mgmt = KEY_MGMT_NONE;
+		bss_config->wpa_cfg.length = 0;
+		priv->sec_info.wep_enabled = 0;
+		priv->sec_info.wpa_enabled = 0;
+		priv->sec_info.wpa2_enabled = 0;
+
+		return 0;
+	}
+
 	switch (params->auth_type) {
 	case NL80211_AUTHTYPE_OPEN_SYSTEM:
 		bss_config->auth_mode = WLAN_AUTH_OPEN;